src/core/ddsc/src/dds_write.c

```diff
@@ -449,8 +453,11 @@ static dds_return_t dds_writecdr_impl_common (struct writer *ddsi_wr, struct nn_
 
 #ifdef DDS_HAS_SHM
   // transfer ownership of an iceoryx chunk if it exists
-  d->iox_chunk = din->iox_chunk;
```

Comment:
This was causing the lost chunks. Maybe this should be part of the `convert_serdata``function instead but I decided against this in the past to not change anything in this function.